File: app-miyou/src/main/java/com/miyou/app/infrastructure/monitoring/repository/SpringDataStagePerformanceRollupRepository.java
package com.miyou.app.infrastructure.monitoring.repository;

import java.time.Instant;

import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.miyou.app.infrastructure.monitoring.document.StagePerformanceRollupDocument;
import reactor.core.publisher.Flux;

public interface SpringDataStagePerformanceRollupRepository
	extends
		ReactiveMongoRepository<StagePerformanceRollupDocument, String> {

	Flux<StagePerformanceRollupDocument> findByGranularityAndBucketStartBetween(
		String granularity,
		Instant startTime,
		Instant endTime);
}

File: backend/monitoring/src/main/kotlin/com/miyou/app/monitoring/repository/SpringDataStagePerformanceRollupRepository.kt
package com.miyou.app.monitoring.repository

import com.miyou.app.monitoring.document.StagePerformanceRollupDocument
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux
import java.time.Instant

interface SpringDataStagePerformanceRollupRepository :
    ReactiveMongoRepository<StagePerformanceRollupDocument, String> {
    fun findByGranularityAndBucketStartBetween(
        granularity: String,
        startTime: Instant,
        endTime: Instant,
    ): Flux<StagePerformanceRollupDocument>
}
